addons/l10n_eg_edi_eta/models/account_edi_format.py

```diff
@@ -0,0 +1,426 @@
+
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        try:
+            request_response = requests.post(
+                request_url, data=data, headers=headers, timeout=(5, 10))
+            submission_data = request_response.json()
+        except (ValueError, requests.exceptions.ConnectionError, requests.exceptions.MissingSchema, requests.exceptions.Timeout, requests.exceptions.HTTPError) as ex:
+            return {invoice: {'error': str(ex), 'blocking_level': 'warning'}}
+
+        if submission_data:
+            if submission_data.get('error'):
+                return {invoice: {
+                    'error': submission_data.get('error'),
+                    'blocking_level': 'error'
+                }}
+            if submission_data.get('rejectedDocuments', False) and isinstance(submission_data.get('rejectedDocuments'), list):
+                return {invoice: {
+                    'error': str(submission_data.get('rejectedDocuments')[0].get('error')),
+                    'blocking_level': 'error'
+                }}
+            if (submission_data.get('submissionId') is not None
+                    and submission_data.get('acceptedDocuments')):
+                eta_invoice_json = json.loads(invoice.l10n_eg_eta_invoice_id.raw)
+                eta_invoice_json['response'] = {
+                    'l10n_eg_uuid': submission_data.get('acceptedDocuments')[0].get('uuid'),
+                    'l10n_eg_long_id': submission_data.get('acceptedDocuments')[0].get('longId'),
+                    'l10n_eg_internal_id': submission_data.get('acceptedDocuments')[0].get('internalId'),
+                    'l10n_eg_hash_key': submission_data.get('acceptedDocuments')[0].get('hashKey'),
+                    'l10n_eg_submission_id': submission_data.get('submissionId'),
+                }
+
+                invoice.l10n_eg_eta_invoice_id.raw = json.dumps(eta_invoice_json)
+                return {invoice: {}}
+
+            return {invoice: {
+                'success': False,
+            }}
+
+    def _cancel_invoice_edi(self, invoices):
+        # OVERRIDE
+        if self.code != 'eg_eta':
+            return super()._cancel_invoice_edi(invoices)
+        invoice = invoices
+        api_domain = self._l10n_eg_get_eta_api_domain()
+        token = self._l10n_eg_get_einvoice_token(invoice)
+        if isinstance(token, dict) and token.get('error', False):
+            return {invoice: token}
+        request_url = '%s/api/v1/documents/state/%s/state' % (
+            api_domain, invoice.l10n_eg_uuid)
+        request_payload = {
+            'status': 'cancelled',
+            'reason': 'Cancelled'
+        }
+        data = json.dumps(request_payload)
+        headers = {'Content-Type': 'application/json',
+                   'Authorization': 'Bearer %s' % token}
+        request_response = requests.request(
```

Comment:
It would be good to catch the right errors here as well.  Should not this be similar as what we have for the sending?  Do we need cancellation at all?  Because credit notes might do the trick as well.